Break the Dedekind reals into lower and upper Dedekind reals (#1314)
UniMath/agda-unimath: src/real-numbers/arithmetically-located-dedekind-cuts.lagda.md

open import foundation.universe-levels

open import group-theory.abelian-groups

open import real-numbers.dedekind-real-numbers
open import real-numbers.lower-dedekind-real-numbers
open import real-numbers.upper-dedekind-real-numbers
```

</details>

## Definition

A pair of a [lower Dedekind real](real-numbers.lower-dedekind-real-numbers.md)
`L` and an [upper Dedekind real](real-numbers.upper-dedekind-real-numbers.md)

> `U` is {{#concept "arithmetically located" Disambiguation="Dedekind cut"}} if

Reviewer: ```suggestion
`U` is {{#concept "arithmetically located" Disambiguation="Dedekind cut" Agda=arithmetically-located-lower-upper-ℝ}} if
```

Author: I imagine this whole section needs to be reworded more clearly in terms of cuts versus lower and upper reals, and I'm working on that, FYI.

for any positive [rational number](elementary-number-theory.rational-numbers.md)
`ε : ℚ`, there exists `p, q : ℚ` where `p ∈ L` and `q ∈ U`, such that
`0 < q - p < ε`. Intuitively, when `L , U` represent the cuts of a real number

> `x`, `p` and `q` are rational approximations of `x` to within `ε`. This follows
> parts of Section 11 in {{#cite BauerTaylor2009}}.

Reviewer: ```suggestion
`x`, `p` and `q` are rational approximations of `x` to within `ε`.

This follows parts of Section 11 in {{#cite BauerTaylor2009}}.
```

## Definitions

### The predicate of being arithmetically located on pairs of subtypes of rational numbers

```agda
module _

>   {l : Level} (L : lower-ℝ l) (U : upper-ℝ l)

Reviewer: ```suggestion
  {l1 l2 : Level} (L : lower-ℝ l1) (U : upper-ℝ l2)
```

open import foundation.universe-levels

open import group-theory.abelian-groups

open import real-numbers.dedekind-real-numbers
open import real-numbers.lower-dedekind-real-numbers
open import real-numbers.upper-dedekind-real-numbers
```

</details>

## Definition

A pair of a [lower Dedekind cut](real-numbers.lower-dedekind-real-numbers.md)
`L` and an [upper Dedekind cut](real-numbers.upper-dedekind-real-numbers.md) `U`
is
{{#concept "arithmetically located" Disambiguation="Dedekind cut" Agda=arithmetically-located-lower-upper-ℝ}}
if for any [positive](elementary-number-theory.positive-rational-numbers.md)
[rational number](elementary-number-theory.rational-numbers.md) `ε : ℚ⁺`, there
exists `p, q : ℚ` where `p ∈ L` and `q ∈ U`, such that `0 < q - p < ε`.

> Intuitively, when `L , U` represent the cuts of a real number `x`, `p` and `q`

Reviewer: ```suggestion
Intuitively, when `L` and `U` are the lower and upper cuts of a real number `x`, `p` and `q`
```

```agda
module _

>   {l : Level} (x : lower-ℝ l) (y : upper-ℝ l)

Reviewer: ```suggestion
  {l1 l2 : Level} (x : lower-ℝ l1) (y : upper-ℝ l2)
```
